internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -26,8 +26,11 @@ import (
 // Max number of times an invalid API Key is checked
 const maxUnauthCounter int = 6
 
-// Const for decraded state or linter complains
-const degraded = "DEGRADED"
+// Consts for states at fleet checkin
+const fleetStateDegraded = "DEGRADED"
+const fleetStateOnline = "online"
+const fleetStateError = "error"
+const fleetStateStarting = "starting"
```

Comment:
Is there some reason we are using all-uppercase for the degraded state and all-lowercase for the other states?

Reply:
I guess this came from a linter issue asking for a constant since "DEGRADED" was used in 2 places, while the other stare are all constant strings used just once in code...
I preferred to have the constants used as output of state in one place but I didn't change the values :sweat_smile: 
But it bugs also my OCD seeing different casing (and also the "screaming" constant values a bit... )

I would love a quick refactor having all the statuses defined as constants and lowercase but maybe not in this bugfix ;)

Reply:
Agreed this reads oddly.

The agent reports status values in upper case in logs and in the output of the `elastic-agent status` command. Users have written scripts to explicitly look for these strings with the upper case values in test automation, deployment automation, etc. We changed the status values to all lowercase originally in 8.6 and then had to restore the upper case versions before release because of all the tests we broke across different teams 🤷 

The the states reported to Fleet are usually in lowercase because this is what the the Fleet UI represents them as internally: https://github.com/elastic/kibana/blob/bb35166fc3b9fdf5d2ab8da9849db0aef253f958/x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_health.tsx#L62-L80. 

I believe the UI calls `.toLowerCase` on these before looking at them so we could change all of these to uppercase if we wanted to.